.pipelines/custom-templates/publish-template.yml

```diff
@@ -0,0 +1,127 @@
+parameters:
+  - name: jobName
+    type: string
+  - name: path
+    type: string
+  - name: libName
+    type: string
+  - name: dependsOn
+    type: string
+    default: 'none'
+  - name: cdn
+    type: boolean
+    default: false
+  - name: publishFlagName
+    type: string
+
+jobs:
+  - job: ${{ parameters.jobName }}
+    displayName: 'Publish ${{ parameters.libName }}'
+    ${{ if not(eq(parameters.dependsOn, 'none')) }}:
+      dependsOn: ${{ parameters.dependsOn }}
+    pool:
+      type: linux
+    variables:
+      repoName: 'msal-javascript-internal'
+      ob_outputDirectory: '${{ parameters.path }}/${{ parameters.libName }}' # this directory is uploaded to pipeline artifacts, reddog and cloudvault. More info at https://aka.ms/obpipelines/artifacts
+      ob_sdl_binskim_break: true # https://aka.ms/obpipelines/sdl
+    condition: eq(variables.publish${{ parameters.publishFlagName }}, true)
+    steps:
+      - task: NodeTool@0
+        displayName: Install Node
+        inputs:
+          versionSpec: '16.x'
+          checkLatest: true
+      # Install dependencies
+      - task: Npm@1
+        displayName: Install dependencies
+        inputs:
+          command: 'install'
+          workingDir: '${{ parameters.path }}/${{ parameters.libName }}'
+          verbose: false
+      # Runs lerna bootstrap scoped to this package and any local dependencies so that npm run build:all commands work
+      - task: Npm@1
+        displayName: Link dependencies
+        inputs:
+          command: 'custom'
+          workingDir: '${{ parameters.path }}/${{ parameters.libName }}'
+          customCommand: 'run link:localDeps --if-present'
+
+  # Publish to CDN
+      # KeyVault connection
+      - task: AzureKeyVault@2
+        inputs:
+          azureSubscription: 'MSIDLABKeyVault'
+          KeyVaultName: 'ADALTestInfo'
+          SecretsFilter: 'MSALJS-CDN-SAS-EUNO, MSALJS-CDN-SAS-USWE, MSALJSNPMTOKEN'
```

Comment:
We will need to make sure that the current tokens are marked as such in the key vault - we have been sloppy about this in the past

Reply:
Not sure exactly what you mean? Do you mean that the names should match?